src/nvim/buffer.c

```diff
@@ -3069,9 +3069,13 @@ static bool ti_change(char_u *str, char_u **last)
 /// Set current window title
 void resettitle(void)
```

Comment:
@nateozem what was the reason for changing this function at all?

Reply:
I made logical reasoning with the old commit from 486d2e9 commit where I follow the logic of `did_set_title`.  From reviewing the changes, I shouldn't place that logic in `resettitle`.  I think because I didn't know if I would have access to variables like `p_title` and `lasttitle`.
My intention were, don't update if there is nothing that needs to be updated.